support/cas-server-support-saml-idp-metadata/src/main/java/org/apereo/cas/support/saml/services/idp/metadata/cache/resolver/MetadataQueryProtocolMetadataResolver.java

```diff
@@ -72,38 +80,38 @@ protected HttpResponse fetchMetadata(final String metadataLocation, final Criter
         return response;
     }
 
-    /**
-     * Is dynamic metadata query configured ?
-     *
-     * @param service the service
-     * @return true/false
-     */
-    protected boolean isDynamicMetadataQueryConfigured(final SamlRegisteredService service) {
-        return service.getMetadataLocation().trim().endsWith("/entities/{0}");
-    }
-
     @Override
     public boolean supports(final SamlRegisteredService service) {
-        return isDynamicMetadataQueryConfigured(service);
+        return SamlUtils.isDynamicMetadataQueryConfigured(service.getMetadataLocation());
     }
 
     @Override
     protected boolean shouldHttpResponseStatusBeProcessed(final HttpStatus status) {
-        return super.shouldHttpResponseStatusBeProcessed(status) || status == HttpStatus.NOT_MODIFIED;
+        return true;
     }
 
     @Override
     protected AbstractMetadataResolver getMetadataResolverFromResponse(final HttpResponse response, final File backupFile) throws Exception {
-        if (response.getStatusLine().getStatusCode() == HttpStatus.NOT_MODIFIED.value()) {
-            return new InMemoryResourceMetadataResolver(backupFile, this.configBean);
+        if (!HttpStatus.valueOf(response.getStatusLine().getStatusCode()).is2xxSuccessful()) {
+            if (Files.exists(backupFile.toPath())) {
+                return new InMemoryResourceMetadataResolver(backupFile, this.configBean);
+            } else {
+                throw new Exception("Unable to get entity from MDQ server and a backup file does not exist.");
+            }
         }
         val entity = response.getEntity();
-        val ins = entity.getContent();
-        val source = ByteStreams.toByteArray(ins);
-        val xmlObject = SamlUtils.transformSamlObject(configBean, source, XMLObject.class);
-        xmlObject.getObjectMetadata().put(new XMLObjectSource(source));
+        val result = IOUtils.toString(entity.getContent(), StandardCharsets.UTF_8);
+        val path = backupFile.toPath();
+        LOGGER.trace("Writing metadata to file at [{}]", path);
+        try (val output = Files.newBufferedWriter(path, StandardCharsets.UTF_8)) {
+            IOUtils.write(result, output);
+            output.flush();
+            output.close();
```

Comment:
This is not necessary. `output` will be automatically closed in the try-resource block.

Reply:
Done